### PR TITLE
[14.0] [NEW] Added new module to restrict chatter buttons.

### PR DESCRIPTION
**Pull request by Chandresh-OSI**
Hide Send Message button from chatter area. (Based on group)

### CODE REVIEW

**Comment by Chandresh-OSI** [comment]
CC @dreispt

File: mail_restrict_send_button/models/mail_followers.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, models
+
+
+class MailFollowers(models.Model):
+    _inherit = "mail.followers"
+
+    @api.model
+    def check_user_group(self):
```
Comment (dreispt) [inline mail_restrict_send_button/models/mail_followers.py]: Method name should be specific about what it does.
```suggestion
    def check_can send_message(self):
```

File: mail_restrict_send_button/static/src/js/chatter.js [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+                values.threadId = clear();
+            }
+            if (!this.chatter) {
+                this.chatter = this.env.models["mail.chatter"].create(values);
+            } else {
+                this.chatter.update(values);
+            }
+        });
+    };
+});
```
Comment (dreispt) [inline mail_restrict_send_button/static/src/js/chatter.js:376]: 400 LoC to add a "isSendMessage" seems a bit too much.
is it possible to not repeat the original source code, and extend just adding the new attribute?

**Review by dreispt — COMMENTED** [review]
LGTM, some comments

File: mail_restrict_send_button/security/res_groups.xml [hunk 3]
```diff
@@ -0,0 +1,7 @@
+<odoo>
+    <data noupdate="1">
+        <record id="group_show_send_message_button" model="res.groups">
+            <field name="name">Show Send Message Button</field>
```
Comment (dreispt) [inline mail_restrict_send_button/security/res_groups.xml:4]: Include this in the "Extra Rights" group (set the application field)

**Review by dreispt — APPROVED** [review]

**Comment by dreispt** [comment]
@Chandresh-OSI Please fix checks

**Review by dreispt — APPROVED** [review]
Tested, working as expected

**Comment by dreispt** [comment]
@Chandresh-OSI An error was found when creating a new Helpdesk Ticket. Can you check please?

With debug assstes on: 
```
Traceback:
Error: clear is not defined
container.prototype._insertFromProps/<@http://testsite.com/mail_restrict_send_button/static/src/js/chatter.js:366:17
```
image

**Comment by clementmbr** [comment]
Hi @dreispt @Chandresh-SerpentCS , 

FYI I tried your module in some complex client base and I have a recurrent (non-blocking but consistent) JS error message  when I create **any** object inheriting mail.thread :

```
Error: scope.threadView.thread is undefined
anonymous@http://172.18.0.2:8069/web/content/2747-1736a5d/web.assets_common.js line 969 > Function:19:13
render@http://172.18.0.2:8069/web/content/2747-1736a5d/web.assets_common.js:958:20
useContextWithCB/__owl__.renderFn@http://172.18.0.2:8069/web/content/2747-1736a5d/web.assets_common.js:1308:113
__render@http://172.18.0.2:8069/web/content/2747-1736a5d/web.assets_common.js:1288:33
render/<@http://172.18.0.2:8069/web/content/2747-1736a5d/web.assets_common.js:1252:6
```
image

It's a non-blocking bug as the object is correctly created even with the error. But the user needs to actualize the page to go ahead.

My problem is that I tried to install this module on an empty base and there **the bug doesn't occur** ! So it might be a conflict with another module but I still don't know which one.

As I'm not a JS expert, I hope you have an idea of where it can comes from or any clue to help me in my research ! :)
Anyway, I let you know if I found something interesting to repeat and avoid this bug.

**Comment by dreispt** [comment]
@clementmbr Maybe same as issue https://github.com/OCA/social/issues/768 ?

**Comment by clementmbr** [comment]
yes! @dreispt 
Perfect! Thanks @auroraeosrose :pray:

**Pull request merged**